src/Components/ServiceScene/Breakdowns/TimeSeriesLimitSeriesTitleItem.tsx

```diff
@@ -0,0 +1,123 @@
+import { Button, Icon, Tooltip, useStyles2 } from '@grafana/ui';
+import React from 'react';
+import { GrafanaTheme2, LoadingState, PanelData } from '@grafana/data';
+import { css } from '@emotion/css';
+import {
+  SceneComponentProps,
+  SceneCSSGridItem,
+  SceneDataTransformer,
+  sceneGraph,
+  SceneObjectBase,
+  SceneObjectState,
+  VizPanel,
+} from '@grafana/scenes';
+
+export const MAX_NUMBER_OF_TIME_SERIES = 20;
+
+export interface TimeSeriesLimitSeriesTitleItemSceneState extends SceneObjectState {
+  toggleShowAllSeries: (model: TimeSeriesLimitSeriesTitleItemScene) => void;
+  showAllSeries: boolean;
+  currentSeriesCount?: number;
+}
+
+export class TimeSeriesLimitSeriesTitleItemScene extends SceneObjectBase<TimeSeriesLimitSeriesTitleItemSceneState> {
+  constructor(state: TimeSeriesLimitSeriesTitleItemSceneState) {
+    super(state);
+
+    this.addActivationHandler(this.onActivate.bind(this));
+  }
+
+  private onActivate() {
+    const panel = sceneGraph.getAncestor(this, VizPanel);
+    this._subs.add(
+      panel.subscribeToState((newState, prevState) => {
+        const $data = sceneGraph.getData(this);
+        if ($data.state.data?.state === LoadingState.Done) {
+          this.setState({
+            currentSeriesCount: $data.state.data?.series.length,
+          });
+        }
+      })
+    );
+  }
+  public static Component = ({ model }: SceneComponentProps<TimeSeriesLimitSeriesTitleItemScene>) => {
+    const { toggleShowAllSeries, showAllSeries, currentSeriesCount } = model.useState();
+    const $data = sceneGraph.getData(model);
+    const { data } = $data.useState();
+    const styles = useStyles2(getStyles);
+
+    if (
+      !($data instanceof SceneDataTransformer) ||
+      showAllSeries ||
+      data?.state !== LoadingState.Done ||
+      !currentSeriesCount ||
+      data.series.length < MAX_NUMBER_OF_TIME_SERIES
+    ) {
+      return null;
+    }
+
+    //@todo is there a better way to get the total number of series before transforming then accessing a private prop?
```

Comment:
This should be "safe enough" for now as we will always run the transformation on dataframe update, so the _prevDataFromSource should always be the untransformed frame. Needs to be tested though if we can't find a better solution